renpy.el

```diff
@@ -1529,6 +1529,109 @@ Uses `renpy-beginning-of-block', `renpy-end-of-block'."
   (renpy-end-of-block)
   (exchange-point-and-mark))
 
+;;;; Completion
+
+(defcustom renpy-setup-completion t
+  "Non-nil means Renpy mode sets up completion for the buffer."
+  :type 'boolean
+  :group 'renpy)
+
+(defun renpy--completion-context ()
+  "Return the completion category for point."
+  ;; TODO: Make sure the point is not within a comment or a string.
+  ;; TODO: More specific contexts: show text, show expression, ...
+  (save-excursion
+    (skip-syntax-backward "w_")
```

Comment:
Is this to move backwards through a partially-typed (prefix) of a completion? If so, what happens when the completion itself has a space in it?
```renpy
image eileen happy = "eileen_happy.png"
```

Reply:
Yes, this is a known broken case currently. Completion will NOT work if called like here:

```renpy
image eileen happy<point>
```

I know it's no good, especially given how Renpy DSL uses spaces a lot. I'll document this as a problem for now and rework things later.

Reply:
It is quite a problem for the default completion interfaces, you can test it by calling `complete-symbol` (key binding C-M-i):

```renpy
image eileen happy = "eileen_happy.png"
image eileen happy other = "eileen_happy_other.png"

show eil<point>
```

The second completion candidate will never be available to the completion, because calling the function once inserts the common prefix. Maybe it would be better to make the completion context for images deal with the image tags and attributes separately instead of trying to complete on the entire definition.

Reply:
I've fixed this by improving prefix autodetection. I'll play with it a bit and see if it works.